sdks/python/apache_beam/transforms/util.py

```diff
@@ -502,8 +502,11 @@ def _calculate_next_batch_size(self):
     target = self._max_batch_size
 
     if self._target_batch_duration_secs:
-      # Solution to a + b*x = self._target_batch_duration_secs.
-      target = min(target, (self._target_batch_duration_secs - a) / b)
+      # Solution to b*x = self._target_batch_duration_secs.
+      # We ignore the fixed cost in this computation as it has negligeabel
```

Comment:
I think this is wrong: what if the element overhead and fixed cost are both large?

Reply:
In that case we still won't batch (much or at all). That's the problem with absolute limits (but it felt too risky of a backwards incompatible change to remove it altogether). But it increases the range that the more general overhead calculation takes precedence instead.